sdk/daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/v2/ledgerinteraction/grpcLedgerClient/GrpcLedgerClient.scala

```
@@ -78,16 +78,44 @@ class GrpcLedgerClient(
 
   // Omits the package id on an identifier if contract upgrades are enabled unless explicitPackageId is true
   private def toApiIdentifierUpgrades(
-      identifier: Identifier,
+      identifier: TypeConRef,
       explicitPackageId: Boolean,
   ): api.Identifier = {
-    val converted = toApiIdentifier(identifier)
+    identifier.pkg match {
+      case PackageRef.Name(name) =>
+        api.Identifier(
+          "#" + name,
+          identifier.qualifiedName.module.toString(),
+          identifier.qualifiedName.name.toString(),
+        )
```

I think error here if `explicitPackageId` is true

---

Right, I misunderstood what was erroring out - we just needed to make sure that Create commands, which have an explicit package ID in their identifier, are treated differently from the Exercise commands that contain a PackageName in their identifier. That's fixed now in the UpgradeMatrixIntegration test.

---

I simply mean for this function, if you were to ever call it with a `NameRef`, but request `explicitPackageId`, then you're clearly calling the function wrong, as such a thing isn't possible. In that case, its nicer to throw an error than silently ignore the flag.